be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -552,6 +564,13 @@ Status BetaRowsetWriter::_close_file_writers() {
         } else {
             RETURN_NOT_OK_STATUS_WITH_WARN(_wait_flying_segcompaction(),
                                            "segcompaction failed when build new rowset");
+            // compact all remaining segments, and since we can't submit a segcompaction
+            // when `_is_doing_segcompaction` is true, wait for the flying segcompaction
+            // to finish first, before doing the check
+            RETURN_NOT_OK_STATUS_WITH_WARN(_segcompaction_if_necessary(true),
```

Comment:
As the 'compact remaining' operation is critical for the close wait latency, queueing in the segcompaction task queue is the last thing we want. So maybe it should have a higher priority than those normal & best-effort segcompaction task?